projects/components/src/download-json/download-json.component.ts

```diff
@@ -0,0 +1,79 @@
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Inject, Input, Renderer2 } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { IconSize } from '@hypertrace/components';
+import { Observable } from 'rxjs';
+import { catchError, finalize, take } from 'rxjs/operators';
+import { ButtonSize, ButtonStyle } from '../button/button';
+import { NotificationService } from '../notification/notification.service';
+
+@Component({
+  selector: 'ht-download-json',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./download-json.component.scss'],
+  template: `
+    <div class="download-json" [htTooltip]="this.tooltip" (click)="this.triggerDownload()">
+      <ht-button
+        *ngIf="!this.dataLoading"
+        class="download-button"
+        icon="${IconType.Download}"
+        display="${ButtonStyle.Text}"
+        size="${ButtonSize.Large}"
+      ></ht-button>
+      <ht-icon *ngIf="this.dataLoading" icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+    </div>
+  `
+})
+export class DownloadJsonComponent {
+  @Input()
+  public dataSource!: Observable<unknown>;
+
+  @Input()
+  public fileName: string = 'download';
+
+  @Input()
+  public tooltip: string = 'Download Json';
+
+  public dataLoading: boolean = false;
+  private readonly dlJsonAnchorElement: HTMLAnchorElement;
+
+  public constructor(
+    @Inject(DOCUMENT) private readonly document: Document,
+    private readonly renderer: Renderer2,
+    private readonly changeDetector: ChangeDetectorRef,
+    private readonly notificationService: NotificationService
+  ) {
+    this.dlJsonAnchorElement = this.document.createElement('a');
```

Comment:
Why not make this an anchor in the template? could we wrap the button in an anchor so the user is actually clicking the anchor they're triggering?

Reply:
Since here we're using an observable to fetch the data, so it is not possible to do that.
We need the data before click, So that's why this is the approach.

Reply:
Even if we don't wrap it (which I still think is doable), having an anchor in this template that's not displayed is more or less equivalent to what we're doing here, so is definitely feasible (basically just changes the createElement to populate that to a `@ViewChild`)